src/sentry/integrations/source_code_management/commit_context.py

```diff
@@ -120,114 +138,137 @@ def queue_comment_task_if_needed(
         group_owner: GroupOwner,
         group_id: int,
     ) -> None:
-        if not OrganizationOption.objects.get_value(
+        with CommitContextIntegrationInteractionEvent(
+            interaction_type=SCMIntegrationInteractionType.QUEUE_COMMENT_TASK,
+            provider_key=self.integration_name,
             organization=project.organization,
-            key="sentry:github_pr_bot",
-            default=True,
-        ):
-            logger.info(
-                _pr_comment_log(integration_name=self.integration_name, suffix="disabled"),
-                extra={"organization_id": project.organization_id},
-            )
-            return
-
-        repo_query = Repository.objects.filter(id=commit.repository_id).order_by("-date_added")
-        group = Group.objects.get_from_cache(id=group_id)
-        if not (
-            group.level is not logging.INFO and repo_query.exists()
-        ):  # Don't comment on info level issues
-            logger.info(
-                _pr_comment_log(
-                    integration_name=self.integration_name, suffix="incorrect_repo_config"
-                ),
-                extra={"organization_id": project.organization_id},
-            )
-            return
-
-        repo: Repository = repo_query.get()
-
-        logger.info(
-            _pr_comment_log(integration_name=self.integration_name, suffix="queue_comment_check"),
-            extra={"organization_id": commit.organization_id, "merge_commit_sha": commit.key},
-        )
-        from sentry.integrations.github.tasks.pr_comment import github_comment_workflow
+            project=project,
+            commit=commit,
+        ).capture() as lifecycle:
+            if not OrganizationOption.objects.get_value(
+                organization=project.organization,
+                key="sentry:github_pr_bot",
+                default=True,
+            ):
+                # TODO: remove logger in favor of the log recorded in  lifecycle.record_halt
+                logger.info(
+                    _pr_comment_log(integration_name=self.integration_name, suffix="disabled"),
+                    extra={"organization_id": project.organization_id},
```

Comment:
i remember we added logging when halts happen. does this mean we can remove these logs?